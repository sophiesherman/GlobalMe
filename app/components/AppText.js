import React from "react";
import { Text, StyleSheet } from "react-native";

import AppColors from "../config/AppColors";
import AppFonts from "../config/AppFonts";

// NAME: AppText: Custom Component

// PURPOSE: one place for a basic text style across the app, style can be overridden if needed

// PARAMS:
//  - children: ensure all things called within an AppText tag will be passed through
//  - style: any styles for the text will be passed through

function AppText({ children, style }) {
  return <Text style={[styles.text, style]}> {children} </Text>;
}

const styles = StyleSheet.create({
  text: {
    fontFamily: AppFonts.body,
    color: AppColors.dark,
  },
});

export default AppText;
