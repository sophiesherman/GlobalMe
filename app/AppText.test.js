import React from "react";
import { Text } from "react-native";
import renderer from "react-test-renderer";
import AppText from "./components/AppText";

test("AppText renders correctly", () => {
  const tree = renderer.create(<AppText />).toJSON();
  expect(tree).toMatchSnapshot();
});

test("AppText is Text component", () => {
  const json = renderer.create(<AppText />).toJSON();
  expect(json.type).toBe("Text");
});

test("AppText contains child text", () => {
  const json = renderer
    .create(<AppText>Some test a user might add</AppText>)
    .toJSON();
  expect(json.children.includes("Some test a user might add"));
});

test("AppText should contain entered style", () => {
  const json = renderer
    .create(<AppText style={{ fontSize: 90 }}>Text</AppText>)
    .toJSON();
  expect(json.props.style[1].fontSize).toBe(90);
});

test("AppText should be the right font family", () => {
  const json = renderer
    .create(<AppText>Some test a user might add</AppText>)
    .toJSON();
  expect(json.props.style[0].fontFamily).toBe("Avenir Next");
});
