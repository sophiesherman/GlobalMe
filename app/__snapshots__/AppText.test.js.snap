// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppText renders correctly 1`] = `
<Text
  style={
    Array [
      Object {
        "color": undefined,
        "fontFamily": "Avenir Next",
      },
      undefined,
    ]
  }
>
   
   
</Text>
`;
